In [1]:
from sklearn.metrics import f1_score, precision_score, recall_score, accuracy_score

In [2]:
def read_train_test_file(name_file):
    # the first row is the number of train_idx and val_idx
    # the second row is the train_idx
    # the third row is the val_idx
    
    with open(name_file, 'r') as f:
        lines = f.readlines()
        num_train, num_val = int(lines[0].split()[0]), int(lines[0].split()[1])
        train_idx = list(map(int, lines[1].split()))
        val_idx = list(map(int, lines[2].split()))
    return num_train, num_val, train_idx, val_idx

num_train, num_val, train_idx, val_idx = read_train_test_file('data\\train_val_group_split.txt')

In [3]:
from sklearn.model_selection import train_test_split
def stratified_split(labels, test_size=0.1, random_state=42):
    indices = list(range(len(labels)))

    # Stratified train/val split
    train_indices, val_indices = train_test_split(
        indices, test_size=test_size, stratify=labels, random_state=random_state
    )

    return train_indices, val_indices

In [4]:
import json
def read_val_submission(name_file):
    # json file
    with open(name_file, 'r') as f:
        data = json.load(f)
    return list(data["results"].values())


In [5]:
import json
def read_labels(name_file):
    # json file 
    with open(name_file,'r') as f:
        data = json.load(f)
    return [data[str(id)]["label"] for id in val_idx],  [data[str(id)]["label"] for id in data.keys()]

In [6]:
labels,full_label = read_labels("data\json\\vimmsd-train.json")

In [7]:
def my_result(file_name,val_test_mode="group"):
    if val_test_mode == "group":
        predictions = read_val_submission(file_name)
        return f1_score(labels,predictions,average="macro")
    if val_test_mode == "stratified":
        predictions = read_val_submission(file_name)
        return f1_score([full_label[i] for i in stratified_split(full_label)[1]],predictions,average="macro")

In [8]:
# test_results = [0.38963485679441767, 0.4077608419746368, 0.3898785810643348, 0.41557615765881695, 0.3787910699255794, 0.38196398763830214]
# models = ["vintern_v1_epoch2","vintern_v1_epoch3","vintern_v2_epoch2","vintern_v2_epoch3", "vintern_v1_epoch4","vintern_v1_epoch5"]
# val_results = [my_result("result_test_val/train_" + name + ".json") for name in models]

In [9]:
test_results = [0.38963485679441767]
models = ["vintern_v1_epoch2"]
val_results = [my_result("result_test_val/train_vintern_v1_epoch2.json")]

In [10]:
test_results += [0.4077608419746368]
models += ["vintern_v1_epoch3"]
val_results +=  [my_result("result_test_val/train_vintern_v1_epoch3.json")]

In [11]:
test_results += [0.3787910699255794]
models += ["vintern_v1_epoch4"]
val_results +=  [my_result("result_test_val/train_vintern_v1_epoch4.json")]

In [12]:
test_results += [0.38196398763830214]
models += ["vintern_v1_epoch5"]
val_results +=  [my_result("result_test_val/train_vintern_v1_epoch5.json")]

In [13]:
test_results += [0.39264108359565886]
models += ["vintern_v1_1_epoch4"]
val_results += [my_result("result_test_val/train_vintern_v1_1_epoch4.json")]

In [14]:
test_results+= [0.397726618]
models += ["vintern_v1_1_epoch5"]
val_results += [my_result("result_test_val/train_vintern_v1_1_epoch5.json")]

In [15]:
test_results += [0.4065629022	]
models += ["vintern_v1_1_epoch6"]
val_results += [my_result("result_test_val/train_vintern_v1_1_epoch6.json")]

In [16]:
test_results += [0.3928831452]
models += ["vintern_v1_1_epoch7"]
val_results += [my_result("result_test_val/train_vintern_v1_1_epoch7.json")]

In [17]:
test_results += [0.3898785810643348]
models += ["vintern_v2_epoch2"]
val_results += [my_result("result_test_val/train_vintern_v2_epoch2.json")]

In [18]:
test_results += [0.41557615765881695]
models += ["vintern_v2_epoch3"]
val_results += [my_result("result_test_val/train_vintern_v2_epoch3.json")]

In [19]:
test_results+= [0.40054043366169567]
models += ["vintern_v2_1_epoch3"]
val_results += [my_result("result_test_val/train_vintern_v2_1_epoch3.json")]

In [20]:
test_results+= [0.397257455]
models += ["vintern_v2_1_epoch4"]
val_results += [my_result("result_test_val/train_vintern_v2_1_epoch4.json")]

In [21]:
test_results+= [0.3920332961]
models += ["vintern_v2_1_epoch5"]
val_results += [my_result("result_test_val/train_vintern_v2_1_epoch5.json")]

In [22]:
test_results+= [0.38716210626827285]
models += ["vintern_v3_epoch3"]
val_results += [my_result("result_test_val/train_vintern_v3_epoch3.json")]

In [23]:
test_results+= [0.3896978576208471]
models += ["vintern_v3_epoch4"]
val_results += [my_result("result_test_val/train_vintern_v3_epoch4.json")]

In [24]:
test_results+= [0.3603096916]
models += ["vintern_v5_epoch7"]
val_results += [my_result("result_test_val/train_vintern_v5_epoch7.json")]

In [25]:
test_results+= [0.3937542693]
models += ["vintern_v6_epoch2"]
val_results += [my_result("result_test_val/train_vintern_v6_epoch2.json")]

In [26]:
test_results+= [0.3857219546]
models += ["vintern_v6_epoch3"]
val_results += [my_result("result_test_val/train_vintern_v6_epoch3.json")]

In [27]:
test_results+= [0.3693553748]
models += ["vintern_v6_1_epoch2"]
val_results += [my_result("result_test_val/train_vintern_v6_1_epoch2.json")]

In [28]:
test_results+= [0.3890042858]
models += ["vintern_v6_1_epoch3"]
val_results += [my_result("result_test_val/train_vintern_v6_1_epoch3.json")]

In [29]:
test_results+= [0.3872594243]
models += ["vintern_v6_1_epoch4"]
val_results += [my_result("result_test_val/train_vintern_v6_1_epoch4.json")]

In [30]:
import pandas as pd 

df = pd.DataFrame({"model":models,"val_f1":val_results,"test_f1":test_results})

In [31]:
def count_sarcasm(file_name):
    with open(file_name, 'r') as f:
        data = json.load(f)
    count_map = {
        "not-sarcasm": 0,
        "multi-sarcasm": 0,
        "text-sarcasm": 0,
        "image-sarcasm": 0
    }
    for label in data["results"].values():
        count_map[label] += 1
    return count_map

In [32]:
# add count_sarcasm to df
pre_fix_train ="train_"
pre_fix_test = "test_"
names = df["model"]
for i in range(len(names)):
    train_name = "result_test_val/"+ pre_fix_train + names[i] + ".json"
    test_name = "result_test_val/"+pre_fix_test + names[i] + ".json"
    df.loc[i,"train_not"] = count_sarcasm(train_name)["not-sarcasm"]
    df.loc[i,"train_multi"] = count_sarcasm(train_name)["multi-sarcasm"]
    df.loc[i,"train_text"] = count_sarcasm(train_name)["text-sarcasm"]
    df.loc[i,"train_image"] = count_sarcasm(train_name)["image-sarcasm"]
    df.loc[i,"test_not"] = count_sarcasm(test_name)["not-sarcasm"]
    df.loc[i,"test_multi"] = count_sarcasm(test_name)["multi-sarcasm"]
    df.loc[i,"test_text"] = count_sarcasm(test_name)["text-sarcasm"]
    df.loc[i,"test_image"] = count_sarcasm(test_name)["image-sarcasm"]

In [33]:
df

model    val_f1   test_f1  train_not  train_multi  \
0     vintern_v1_epoch2  0.396414  0.389635      875.0        738.0   
1     vintern_v1_epoch3  0.395101  0.407761      913.0        695.0   
2     vintern_v1_epoch4  0.391509  0.378791      914.0        682.0   
3     vintern_v1_epoch5  0.388991  0.381964      904.0        670.0   
4   vintern_v1_1_epoch4  0.395932  0.392641      916.0        676.0   
5   vintern_v1_1_epoch5  0.390820  0.397727      935.0        659.0   
6   vintern_v1_1_epoch6  0.391938  0.406563      938.0        654.0   
7   vintern_v1_1_epoch7  0.395888  0.392883      935.0        653.0   
8     vintern_v2_epoch2  0.396835  0.389879      889.0        723.0   
9     vintern_v2_epoch3  0.385083  0.415576      932.0        670.0   
10  vintern_v2_1_epoch3  0.388734  0.400540      892.0        713.0   
11  vintern_v2_1_epoch4  0.386626  0.397257      909.0        695.0   
12  vintern_v2_1_epoch5  0.385047  0.392033      914.0        691.0   
13    vintern_v3_epoch3  0.400583  0.387162      891.0        722.0   
14    vintern_v3_epoch4  0.399171  0.389698      867.0        747.0   
15    vintern_v5_epoch7  0.405613  0.360310      918.0        683.0   
16    vintern_v6_epoch2  0.391073  0.393754      869.0        743.0   
17    vintern_v6_epoch3  0.388789  0.385722      907.0        703.0   
18  vintern_v6_1_epoch2  0.381459  0.369355      882.0        732.0   
19  vintern_v6_1_epoch3  0.387275  0.389004      891.0        721.0   
20  vintern_v6_1_epoch4  0.382539  0.387259      910.0        700.0   

    train_text  train_image  test_not  test_multi  test_text  test_image  
0          0.0          6.0     473.0       939.0        0.0         1.0  
1          0.0         11.0     487.0       917.0        0.0         9.0  
2          0.0         23.0     517.0       881.0        0.0        15.0  
3          1.0         44.0     533.0       845.0        0.0        35.0  
4          0.0         27.0     524.0       874.0        0.0        15.0  
5          0.0         25.0     534.0       858.0        0.0        21.0  
6          0.0         27.0     539.0       853.0        0.0        21.0  
7          0.0         31.0     539.0       849.0        0.0        25.0  
8          0.0          7.0     460.0       950.0        0.0         3.0  
9          0.0         17.0     502.0       902.0        0.0         9.0  
10         0.0         14.0     484.0       923.0        0.0         6.0  
11         0.0         15.0     498.0       908.0        0.0         7.0  
12         0.0         14.0     506.0       899.0        0.0         8.0  
13         0.0          6.0     445.0       966.0        0.0         2.0  
14         0.0          5.0     434.0       978.0        0.0         1.0  
15         0.0         18.0     534.0       873.0        0.0         6.0  
16         0.0          7.0     452.0       960.0        0.0         1.0  
17         0.0          9.0     496.0       913.0        0.0         4.0  
18         0.0          5.0     472.0       941.0        0.0         0.0  
19         0.0          7.0     470.0       942.0        0.0         1.0  
20         0.0          9.0     483.0       927.0        0.0         3.0

In [34]:
df[["model","train_not","train_image","train_multi","test_not","test_image","test_multi","val_f1","test_f1"]]

model  train_not  train_image  train_multi  test_not  \
0     vintern_v1_epoch2      875.0          6.0        738.0     473.0   
1     vintern_v1_epoch3      913.0         11.0        695.0     487.0   
2     vintern_v1_epoch4      914.0         23.0        682.0     517.0   
3     vintern_v1_epoch5      904.0         44.0        670.0     533.0   
4   vintern_v1_1_epoch4      916.0         27.0        676.0     524.0   
5   vintern_v1_1_epoch5      935.0         25.0        659.0     534.0   
6   vintern_v1_1_epoch6      938.0         27.0        654.0     539.0   
7   vintern_v1_1_epoch7      935.0         31.0        653.0     539.0   
8     vintern_v2_epoch2      889.0          7.0        723.0     460.0   
9     vintern_v2_epoch3      932.0         17.0        670.0     502.0   
10  vintern_v2_1_epoch3      892.0         14.0        713.0     484.0   
11  vintern_v2_1_epoch4      909.0         15.0        695.0     498.0   
12  vintern_v2_1_epoch5      914.0         14.0        691.0     506.0   
13    vintern_v3_epoch3      891.0          6.0        722.0     445.0   
14    vintern_v3_epoch4      867.0          5.0        747.0     434.0   
15    vintern_v5_epoch7      918.0         18.0        683.0     534.0   
16    vintern_v6_epoch2      869.0          7.0        743.0     452.0   
17    vintern_v6_epoch3      907.0          9.0        703.0     496.0   
18  vintern_v6_1_epoch2      882.0          5.0        732.0     472.0   
19  vintern_v6_1_epoch3      891.0          7.0        721.0     470.0   
20  vintern_v6_1_epoch4      910.0          9.0        700.0     483.0   

    test_image  test_multi    val_f1   test_f1  
0          1.0       939.0  0.396414  0.389635  
1          9.0       917.0  0.395101  0.407761  
2         15.0       881.0  0.391509  0.378791  
3         35.0       845.0  0.388991  0.381964  
4         15.0       874.0  0.395932  0.392641  
5         21.0       858.0  0.390820  0.397727  
6         21.0       853.0  0.391938  0.406563  
7         25.0       849.0  0.395888  0.392883  
8          3.0       950.0  0.396835  0.389879  
9          9.0       902.0  0.385083  0.415576  
10         6.0       923.0  0.388734  0.400540  
11         7.0       908.0  0.386626  0.397257  
12         8.0       899.0  0.385047  0.392033  
13         2.0       966.0  0.400583  0.387162  
14         1.0       978.0  0.399171  0.389698  
15         6.0       873.0  0.405613  0.360310  
16         1.0       960.0  0.391073  0.393754  
17         4.0       913.0  0.388789  0.385722  
18         0.0       941.0  0.381459  0.369355  
19         1.0       942.0  0.387275  0.389004  
20         3.0       927.0  0.382539  0.387259

In [35]:


new_col = "test_not_epoch_2"
new_col_2 = "train_not_epoch_2"
# first fill with nan
df[new_col] = None
df[new_col_2] = None

In [36]:
# best_v1_epoch2_not_v1_epoch2  0.3950902131408822 v1_1_epoch2 0.3877365755677894    
# best_v1_epoch3_not_v1_epoch2  0.4184862223145145 v1_1_epoch2  0.41112557205635963
# best_v1_epoch4_not_v1_epoch2  0.40056369575091666 v1_1_epoch2  0.39354356584004613
# best_v1_epoch5_not_v1_epoch2  0.4088158235206607 v1_1_epoch2  0.4015181636627898
# best_v1_1_epoch4_not_v1_epoch2  0.41245013426850885 v1_1_epoch2  0.4054285422255307

# best_v2_epoch2_not_v1_epoch2 0.3936808317887903 v1_1_epoch2 0.38632708878568167
# best_v2_epoch3_not_v1_epoch2 0.4177265599915593 v1_1_epoch2 0.4108691617999494
# best_v2_1_epoch3_not_v1_epoch2  0.4071201253687884 v1_1_epoch2  0.39976042914937293

df.loc[df["model"] == "vintern_v1_epoch2",new_col] = 0.3950902131408822
df.loc[df["model"] == "vintern_v1_epoch3",new_col] = 0.4184862223145145
df.loc[df["model"] == "vintern_v1_epoch4",new_col] = 0.40056369575091666
df.loc[df["model"] == "vintern_v1_epoch5",new_col] = 0.4088158235206607

df.loc[df["model"] == "vintern_v1_1_epoch4",new_col] = 0.41245013426850885
df.loc[df["model"] == "vintern_v1_1_epoch5",new_col] = 0.4177417747	
df.loc[df["model"] == "vintern_v1_1_epoch6",new_col] = 0.4282051925	
df.loc[df["model"] == "vintern_v1_1_epoch7",new_col] = 0.4145782666	

df.loc[df["model"] == "vintern_v2_epoch2",new_col] = 0.3936808317887903
df.loc[df["model"] == "vintern_v2_epoch3",new_col] = 0.4177265599915593

df.loc[df["model"] == "vintern_v2_1_epoch3",new_col] = 0.4071201253687884
df.loc[df["model"] == "vintern_v2_1_epoch4",new_col] = 0.4060422571	
df.loc[df["model"] == "vintern_v2_1_epoch5",new_col] = 0.4050995119	

df.loc[df["model"] == "vintern_v3_epoch3",new_col] = 0.394623865
df.loc[df["model"] == "vintern_v3_epoch4",new_col] = 0.3950902131

df.loc[df["model"] == "vintern_v6_epoch2",new_col] = 0.3950902131
df.loc[df["model"] == "vintern_v6_epoch3",new_col] = 0.3930280533	

df.loc[df["model"] == "vintern_v6_1_epoch2",new_col] = 0.3777405806	
df.loc[df["model"] == "vintern_v6_1_epoch3",new_col] = 0.3950902131	
df.loc[df["model"] == "vintern_v6_1_epoch4",new_col] = 0.3936808318	


In [37]:
for name in models:
    name_best1 = name[8:]
    name_not = "v1_epoch2"
    df.loc[df["model"] == name,new_col_2] = my_result(f"merge_val/best_not/val_best_{name_best1}_not_{name_not}_results.json")

In [38]:
df[["model",new_col,new_col_2,"test_f1","val_f1"]]

model test_not_epoch_2 train_not_epoch_2   test_f1    val_f1
0     vintern_v1_epoch2          0.39509          0.403039  0.389635  0.396414
1     vintern_v1_epoch3         0.418486          0.406721  0.407761  0.395101
2     vintern_v1_epoch4         0.400564          0.410996  0.378791  0.391509
3     vintern_v1_epoch5         0.408816          0.413382  0.381964  0.388991
4   vintern_v1_1_epoch4          0.41245          0.414021  0.392641  0.395932
5   vintern_v1_1_epoch5         0.417742          0.409157  0.397727  0.390820
6   vintern_v1_1_epoch6         0.428205          0.414009  0.406563  0.391938
7   vintern_v1_1_epoch7         0.414578          0.417392  0.392883  0.395888
8     vintern_v2_epoch2         0.393681          0.402125  0.389879  0.396835
9     vintern_v2_epoch3         0.417727           0.40254  0.415576  0.385083
10  vintern_v2_1_epoch3          0.40712          0.404749  0.400540  0.388734
11  vintern_v2_1_epoch4         0.406042          0.403877  0.397257  0.386626
12  vintern_v2_1_epoch5           0.4051          0.404749  0.392033  0.385047
13    vintern_v3_epoch3         0.394624          0.402687  0.387162  0.400583
14    vintern_v3_epoch4          0.39509          0.403624  0.389698  0.399171
15    vintern_v5_epoch7             None          0.422636  0.360310  0.405613
16    vintern_v6_epoch2          0.39509          0.393928  0.393754  0.391073
17    vintern_v6_epoch3         0.393028          0.400766  0.385722  0.388789
18  vintern_v6_1_epoch2         0.377741          0.386675  0.369355  0.381459
19  vintern_v6_1_epoch3          0.39509          0.393928  0.389004  0.387275
20  vintern_v6_1_epoch4         0.393681          0.392477  0.387259  0.382539

In [39]:
new_col = "te_b2_1epoch5_no_epoch2"
new_col_2 = "tr_b2_1epoch5_no_epoch2"
# first fill with nan
df[new_col] = None
df[new_col_2] = None

In [40]:
df.loc[df["model"] == "vintern_v1_epoch2",new_col] = 0.4097484303
df.loc[df["model"] == "vintern_v1_epoch3",new_col] = 0.4178788014
df.loc[df["model"] == "vintern_v1_epoch4",new_col] = 0.4100418115
df.loc[df["model"] == "vintern_v1_epoch5",new_col] = 0.4088158235

df.loc[df["model"] == "vintern_v1_1_epoch4",new_col] = 0.4150220253
df.loc[df["model"] == "vintern_v1_1_epoch5",new_col] = 0.4126274006
df.loc[df["model"] == "vintern_v1_1_epoch6",new_col] = 0.4200827755
df.loc[df["model"] == "vintern_v1_1_epoch7",new_col] = 0.4123917101

df.loc[df["model"] == "vintern_v2_epoch2",new_col] = 0.4156061973
df.loc[df["model"] == "vintern_v2_epoch3",new_col] = 0.4231450647

df.loc[df["model"] == "vintern_v2_1_epoch3",new_col] = 0.4156246152
df.loc[df["model"] == "vintern_v2_1_epoch4",new_col] = 0.416014107
df.loc[df["model"] == "vintern_v2_1_epoch5",new_col] = 0.4139232514

df.loc[df["model"] == "vintern_v3_epoch3",new_col] = 0.411874495
df.loc[df["model"] == "vintern_v3_epoch4",new_col] = 0.4162646894

df.loc[df["model"] == "vintern_v5_epoch7",new_col] = 0.4046812355

df.loc[df["model"] == "vintern_v6_epoch2",new_col] = 0.4187125924
df.loc[df["model"] == "vintern_v6_epoch3",new_col] = 0.4179877155	

df.loc[df["model"] == "vintern_v6_1_epoch2",new_col] = 0.409123399
df.loc[df["model"] == "vintern_v6_1_epoch3",new_col] = 0.4169328012	
df.loc[df["model"] == "vintern_v6_1_epoch4",new_col] = 0.4183148058


In [41]:
name_best2 = "v1_epoch5"
for name in models:
    name_best1 = name[8:]
    name_not = "v1_epoch2"
    df.loc[df["model"] == name,new_col_2] = my_result(f"merge_val/best1_best2_not/val_best_{name_best1}_best2_{name_best2}_not_{name_not}_results.json")

In [42]:
df[["model","te_b2_1epoch5_no_epoch2","tr_b2_1epoch5_no_epoch2","test_not_epoch_2","train_not_epoch_2"]]

model te_b2_1epoch5_no_epoch2 tr_b2_1epoch5_no_epoch2  \
0     vintern_v1_epoch2                0.409748                0.413828   
1     vintern_v1_epoch3                0.417879                0.414609   
2     vintern_v1_epoch4                0.410042                0.411074   
3     vintern_v1_epoch5                0.408816                0.413382   
4   vintern_v1_1_epoch4                0.415022                0.414375   
5   vintern_v1_1_epoch5                0.412627                0.410585   
6   vintern_v1_1_epoch6                0.420083                0.415078   
7   vintern_v1_1_epoch7                0.412392                0.414509   
8     vintern_v2_epoch2                0.415606                0.414567   
9     vintern_v2_epoch3                0.423145                0.409662   
10  vintern_v2_1_epoch3                0.415625                0.412156   
11  vintern_v2_1_epoch4                0.416014                0.411377   
12  vintern_v2_1_epoch5                0.413923                0.411106   
13    vintern_v3_epoch3                0.411874                0.415646   
14    vintern_v3_epoch4                0.416265                0.413843   
15    vintern_v5_epoch7                0.404681                0.420228   
16    vintern_v6_epoch2                0.418713                0.413844   
17    vintern_v6_epoch3                0.417988                0.413325   
18  vintern_v6_1_epoch2                0.409123                0.413542   
19  vintern_v6_1_epoch3                0.416933                0.413834   
20  vintern_v6_1_epoch4                0.418315                0.411854   

   test_not_epoch_2 train_not_epoch_2  
0           0.39509          0.403039  
1          0.418486          0.406721  
2          0.400564          0.410996  
3          0.408816          0.413382  
4           0.41245          0.414021  
5          0.417742          0.409157  
6          0.428205          0.414009  
7          0.414578          0.417392  
8          0.393681          0.402125  
9          0.417727           0.40254  
10          0.40712          0.404749  
11         0.406042          0.403877  
12           0.4051          0.404749  
13         0.394624          0.402687  
14          0.39509          0.403624  
15             None          0.422636  
16          0.39509          0.393928  
17         0.393028          0.400766  
18         0.377741          0.386675  
19          0.39509          0.393928  
20         0.393681          0.392477

In [43]:
new_col = "te_b2_1epoch5_b3_1.1_e7_no_e2"
new_col_2 = "tr_b2_1epoch5_b3_1.1_e7_no_e2"
# first fill with nan
df[new_col] = None
df[new_col_2] = None
df.loc[df["model"] == "vintern_v1_epoch2",new_col] = 0.4059251449
df.loc[df["model"] == "vintern_v1_epoch3",new_col] = 0.4272139271
df.loc[df["model"] == "vintern_v1_epoch4",new_col] = 0.4068554309
# df.loc[df["model"] == "vintern_v1_epoch5",new_col] = 0.4088158235

df.loc[df["model"] == "vintern_v1_1_epoch4",new_col] = 0.4228889042
df.loc[df["model"] == "vintern_v1_1_epoch5",new_col] = 0.4176994417
df.loc[df["model"] == "vintern_v1_1_epoch6",new_col] = 0.419226467
df.loc[df["model"] == "vintern_v1_1_epoch7",new_col] = 0.4151547086

df.loc[df["model"] == "vintern_v2_epoch2",new_col] = 0.402652293
df.loc[df["model"] == "vintern_v2_epoch3",new_col] = 0.399597092

df.loc[df["model"] == "vintern_v2_1_epoch3",new_col] = 0.4017954055
df.loc[df["model"] == "vintern_v2_1_epoch4",new_col] = 0.4011098206
df.loc[df["model"] == "vintern_v2_1_epoch5",new_col] = 0.3997849423

df.loc[df["model"] == "vintern_v3_epoch3",new_col] = 0.3908558732
df.loc[df["model"] == "vintern_v3_epoch4",new_col] = 0.3916263655

df.loc[df["model"] == "vintern_v5_epoch7",new_col] = 0.3768483227

df.loc[df["model"] == "vintern_v6_epoch2",new_col] = 0.4025617712
df.loc[df["model"] == "vintern_v6_epoch3",new_col] = 0.4002204913

df.loc[df["model"] == "vintern_v6_1_epoch2",new_col] = 0.3901140964
df.loc[df["model"] == "vintern_v6_1_epoch3",new_col] = 0.4016438128
df.loc[df["model"] == "vintern_v6_1_epoch4",new_col] =0.3999612628


In [44]:
name_best2 = "v1_epoch5"
name_best3 = "v1_1_epoch7"
for name in models:
    name_best1 = name[8:]
    name_not = "v1_epoch2"
    df.loc[df["model"] == name,new_col_2] = my_result(f"merge_val/best1_best2_best3_not/val_best_{name_best1}_best2_{name_best2}_best3_{name_best3}_not_{name_not}_results.json")

In [45]:
df[["model","te_b2_1epoch5_b3_1.1_e7_no_e2","tr_b2_1epoch5_b3_1.1_e7_no_e2","te_b2_1epoch5_no_epoch2","tr_b2_1epoch5_no_epoch2","test_not_epoch_2","train_not_epoch_2"]]

model te_b2_1epoch5_b3_1.1_e7_no_e2  \
0     vintern_v1_epoch2                      0.405925   
1     vintern_v1_epoch3                      0.427214   
2     vintern_v1_epoch4                      0.406855   
3     vintern_v1_epoch5                          None   
4   vintern_v1_1_epoch4                      0.422889   
5   vintern_v1_1_epoch5                      0.417699   
6   vintern_v1_1_epoch6                      0.419226   
7   vintern_v1_1_epoch7                      0.415155   
8     vintern_v2_epoch2                      0.402652   
9     vintern_v2_epoch3                      0.399597   
10  vintern_v2_1_epoch3                      0.401795   
11  vintern_v2_1_epoch4                       0.40111   
12  vintern_v2_1_epoch5                      0.399785   
13    vintern_v3_epoch3                      0.390856   
14    vintern_v3_epoch4                      0.391626   
15    vintern_v5_epoch7                      0.376848   
16    vintern_v6_epoch2                      0.402562   
17    vintern_v6_epoch3                       0.40022   
18  vintern_v6_1_epoch2                      0.390114   
19  vintern_v6_1_epoch3                      0.401644   
20  vintern_v6_1_epoch4                      0.399961   

   tr_b2_1epoch5_b3_1.1_e7_no_e2 te_b2_1epoch5_no_epoch2  \
0                       0.414856                0.409748   
1                       0.415853                0.417879   
2                       0.406344                0.410042   
3                       0.398935                0.408816   
4                       0.419918                0.415022   
5                        0.41619                0.412627   
6                       0.420855                0.420083   
7                       0.419276                0.412392   
8                       0.421563                0.415606   
9                       0.421503                0.423145   
10                      0.416217                0.415625   
11                      0.417781                0.416014   
12                      0.422488                0.413923   
13                       0.42203                0.411874   
14                      0.422862                0.416265   
15                      0.423497                0.404681   
16                      0.410279                0.418713   
17                      0.412511                0.417988   
18                      0.404935                0.409123   
19                      0.415051                0.416933   
20                      0.414131                0.418315   

   tr_b2_1epoch5_no_epoch2 test_not_epoch_2 train_not_epoch_2  
0                 0.413828          0.39509          0.403039  
1                 0.414609         0.418486          0.406721  
2                 0.411074         0.400564          0.410996  
3                 0.413382         0.408816          0.413382  
4                 0.414375          0.41245          0.414021  
5                 0.410585         0.417742          0.409157  
6                 0.415078         0.428205          0.414009  
7                 0.414509         0.414578          0.417392  
8                 0.414567         0.393681          0.402125  
9                 0.409662         0.417727           0.40254  
10                0.412156          0.40712          0.404749  
11                0.411377         0.406042          0.403877  
12                0.411106           0.4051          0.404749  
13                0.415646         0.394624          0.402687  
14                0.413843          0.39509          0.403624  
15                0.420228             None          0.422636  
16                0.413844          0.39509          0.393928  
17                0.413325         0.393028          0.400766  
18                0.413542         0.377741          0.386675  
19                0.413834          0.39509          0.393928  
20                0.411854         0.393681          0.392477

vintern_v1_epoch3 
+ te_b2_1epoch5_no_epoch2: 	0.417879
+ tr_b2_1epoch5_no_epoch2: 0.414609

vintern_v1_1_epoch4
+ te_b2_1epoch5_b3_1.1_e7_no_e2: 0.422889
+ tr_b2_1epoch5_b3_1.1_e7_no_e2: 0.419918	

vintern_v1_1_epoch5		
+ te_b2_1epoch5_b3_1.1_e7_no_e2: 0.417699
+ tr_b2_1epoch5_b3_1.1_e7_no_e2: 0.41619	

vintern_v1_1_epoch6
+ te_b2_1epoch5_b3_1.1_e7_no_e2: 0.419226	
+ tr_b2_1epoch5_b3_1.1_e7_no_e2: 0.420855

vintern_v2_epoch2
+ te_b2_1epoch5_no_epoch2: 	0.415606	
+ tr_b2_1epoch5_no_epoch2: 0.414567

In [46]:
# save to model.csv

df.to_csv("model.csv",index=False)
